contrib/checkstyle/src/mill/contrib/checkstyle/CheckstyleModule.scala

```diff
@@ -0,0 +1,184 @@
+package mill.contrib.checkstyle
+
+import com.etsy.sbt.checkstyle.CheckstyleSeverityLevel.CheckstyleSeverityLevel
+import com.etsy.sbt.checkstyle.{
+  CheckstyleConfigLocation,
+  CheckstyleSeverityLevel,
+  CheckstyleXSLTSettings
+}
+import mill.api.Logger
+import mill.{Agg, T}
+import mill.scalalib.{Dep, DepSyntax, JavaModule}
+import mill.util.Jvm
+import net.sf.saxon.s9api.Processor
+import os.Path
+
+import java.io.File
+import javax.xml.transform.stream.StreamSource
+import scala.xml.{Elem, SAXParser}
+import scala.xml.factory.XMLLoader
+
+/**
+ * Integrate Checkstyle into a [[JavaModule]].
+ *
+ * See https://checkstyle.sourceforge.io/
+ */
+trait CheckstyleModule extends JavaModule {
+
+  /** The `checkstyle` version to use. Defaults to `10.18.1`. */
+  def checkstyleVersion: T[String] = T.input { "10.18.1" }
+
+  /** The location of the `checkstyle` configuration file */
+  def checkstyleConfigLocation: T[CheckstyleConfigLocation] = T.input {
+    CheckstyleConfigLocation.File((T.workspace / "checkstyle-config.xml").toString())
+  }
+
+  /** An optional set of XSLT transformations to be applied to the checkstyle output */
+  def checkstyleXsltTransformations: T[Option[Set[CheckstyleXSLTSettings]]] = T.input { None }
```

Comment:
This type signature here looks a bit off. Do we need it to be wrapped in `Option`, when it could be a `Set.empty`? If transforms are to be run sequentially, and the ordering matters, should it be a `Seq` instead?

Reply:
Agreed that there's no point in the `Option` wrapping, will fix that.

I think this is a pretty niche feature so probably the default should be no such file.